src/cider/nrepl/middleware/info.clj

```diff
@@ -109,6 +109,14 @@
            (if resource-relative
              {:resource resource-relative}))))
 
+(defmacro with-java-api-version
```

Comment:
Or we can just make some tests Java version specific, which might be better. This approach seems a bit ugly to me, although it makes sense.

Reply:
Actually I don't think this is possible because `cloverage` doesn't support test selectors (see cloverage/cloverage#54). Cloverage has other mechanisms like filtering test namespaces by regex but that would also be ugly.

Reply:
One other possibility I'm looking at is `circleci/circleci.test` as a substitute for the default leiningen test runner which is a "massive and terrible monkeypatch" (their words, not mine).  The main thing is that circleci.test re-implements the test selector logic and is already integrated with cloverage. The missing piece is that cloverage can't/doesn't have a command line flag to pass in a selector keyword. Which would be a pull request in cloverage. If you want to wait for that, we could try for that route.

Reply:
I pushed a change that uses commons-lang3, which is a dependency leiningen already uses, to mark tests that need to run in specific jvm versions.

```
(if (SystemUtils/IS_JAVA_1_8)
  (testing "a Java 8 specific test case"
    (is (= expected (run-test)))))
```